### ISSUE TITLE
False positive invalid-name (C0103) for module-level TypedDict instantiation, regression from PR #10673

### ISSUE THREAD

**Issue by v-startsev** [PyCQA/pylint#11231]
### Bug description

After #10672 was fixed by #10673, pylint now flags `invalid-name` (C0103) for *any* module-level
variable assigned an **instance** of a `TypedDict`-derived class — not just for the functional
`typing.TypedDict("Name", {...})` type-definition syntax that #10672 was actually about.

`_should_check_class_regex()` treats an assignment as class-like whenever the inferred value is
a `bases.Instance` whose MRO contains an ancestor named `"TypedDict"`:

```python
if isinstance(inferred_assign_type, bases.Instance) and {
    "EnumMeta",
    "TypedDict",
}.intersection(
    {ancestor.name for ancestor in cast(InferenceResult, inferred_assign_type).mro()}
):
    return True
```

This is correct for the functional-syntax case (`X = typing.TypedDict("X", {...})`), where `X`
really does define a new type and should follow `class-rgx`. But it also fires for the completely
unrelated case of instantiating an *already-defined* TypedDict subclass to produce an ordinary
value — which is not a type definition at all and shouldn't be checked against `class-rgx`.

In the type-definition case, astroid already infers the assigned value as a `nodes.ClassDef`,
which is caught by the *first* branch of `_should_check_class_regex` (`isinstance(inferred_assign_type, nodes.ClassDef)`).
The second branch (the `Instance`/MRO check added by #10673) is only ever needed for the `EnumMeta`
case, not for `TypedDict`, and adding `"TypedDict"` to that same branch overreaches into plain
value construction.

### Minimal reproducer

```python
from typing import TypedDict


class Foo(TypedDict):
    value: str


_DEFAULT_VALUE = Foo(value='0')
```

### Configuration

```ini

```

### Command used

```shell
pylint --disable=all --enable=invalid-name repro.py
```

### Pylint output

```python
************* Module repro
repro.py:8:0: C0103: Class name "_DEFAULT_VALUE" doesn't conform to PascalCase naming style (invalid-name)
```

### Expected behavior

No `invalid-name` warning — `_DEFAULT_VALUE` is an ordinary module-level constant holding a plain `dict`-like value (`Foo` is a `dict` at runtime), not a class or a type definition. It should be checked against `const-rgx`/`variable-rgx`, not `class-rgx`.

### Pylint version

```shell
pylint 4.0.6
astroid 4.0.4
Python 3.14.6
```

### OS / Environment

Fedora 44

### Additional dependencies

```python

```
